pkg/controller/endpointslicemirroring/utils.go

```diff
@@ -278,3 +290,35 @@ func hasLeaderElection(annotations map[string]string) bool {
 	_, ok := annotations[resourcelock.LeaderElectionRecordAnnotationKey]
 	return ok
 }
+
+// updateEndpointSliceLabels returns true if all labels in a are not present in b
+func updateEndpointSliceLabels(a, b map[string]string) bool {
+	if len(a) > len(b) {
+		return true
+	}
+	if reflect.DeepEqual(a, b) {
+		return false
+	}
+	for k, v1 := range a {
+		if v2, ok := b[k]; !ok || v1 != v2 {
+			return true
+		}
+	}
+	return false
+}
+
+// updateEndpointSliceAnnotations returns true if the annotations in a and b are
+// not equal (skipping the corev1.EndpointsLastChangeTriggerTime annotation in a)
+func updateEndpointSliceAnnotations(a, b map[string]string) bool {
+	annotations := map[string]string{}
+	for k, v := range a {
+		if k == corev1.EndpointsLastChangeTriggerTime {
+			continue
+		}
+		annotations[k] = v
+	}
+	if !apiequality.Semantic.DeepEqual(annotations, b) {
```

Comment:
Why the difference in `DeepEqual` calls here and above? (`apiequality` vs `reflect`)

Reply:
can't remember, unintentionally for sure

Reply:
we need to use apiequality because an empty map and nil are the same for us